src/test/java/com/atlassian/db/replica/api/TestDualConnection.java

```diff
@@ -1791,4 +1792,39 @@ public void shouldPropagateLastWrite() throws SQLException {
 
         assertThat(lastWriteCache.get()).isPresent();
     }
+
+    @Test
+    public void shouldNotCallDirtyConnectionHookAfterARollback() throws SQLException {
+        final ConnectionProviderMock provider = getProviderThrowingExceptionsOnConnectionRollback();
+        final DirtyConnectionCloseHook dirtyConnectionCloseHook = mock(DirtyConnectionCloseHook.class);
+        final Connection connection = DualConnection.builder(provider, permanentInconsistency().build())
+            .dirtyConnectionCloseHook(dirtyConnectionCloseHook)
+            .build();
+
+        connection.setAutoCommit(false);
```

Comment:
Wait, shouldn't jira has enabled autoCommit? 
So there are cases when rollback happen, connection is shortly close afterwards and.. what should happen with connection? It shouldn't be committed, that's why you assert on:
```
 verify(dirtyConnectionCloseHook, never()).onClose(any(Connection.class));
```
?
and those false positives... it means that there is no a problem at all, right?

Reply:
>So there are cases when rollback happen, connection is shortly close afterwards and.. what should happen with connection? It shouldn't be committed

I don't understand it. The transaction was rolled back and the connection closes. Everything seems to work as designed.